### PR TITLE
Async Analytics Provider

### PR DESCRIPTION
**Pull request by grouma**
Move the async work required for the remote provider into `main.dart` so there isn't potentially a flash with the `AnalyticsPrompt`.

### CODE REVIEW

File: packages/devtools_app/lib/src/analytics/prompt.dart [hunk 1]
```diff
@@ -36,23 +36,19 @@ class _AnalyticsPromptState extends State<AnalyticsPrompt> {
...
-            setState(() {
-              _isVisible = true;
-            });
-          } else if (await _provider.isEnabled) {
-            _provider.setUpAnalytics();
-          }
+    try {
+      if (_provider.isGtagsEnabled) {
+        if (_provider.isFirstRun) {
+          setState(() {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/analytics/prompt.dart:42]: don't need a setState within initState.

File: packages/devtools_app/lib/src/analytics/prompt.dart [hunk 2]
```diff
@@ -36,23 +36,19 @@ class _AnalyticsPromptState extends State<AnalyticsPrompt> {
...
+          });
+        } else if (_provider.isEnabled) {
+          _provider.setUpAnalytics();
         }
-      } catch (e) {
-        print('Error collecting analytics:\n$e');
       }
-    }();
+    } catch (e) {
+      print('Error collecting analytics:\n$e');
```
Comment (jacob314) [inline packages/devtools_app/lib/src/analytics/prompt.dart:50]: switch this print to log and/or remove.

File: packages/devtools_app/lib/src/app.dart [hunk 3]
```diff
@@ -46,11 +47,17 @@ const codeSizeRoute = '/codeSize';
 /// Top-level configuration for the app.
 @immutable
 class DevToolsApp extends StatefulWidget {
-  const DevToolsApp(this.screens, this.preferences, this.ideTheme);
+  const DevToolsApp(
+    this.screens,
+    this.preferences,
+    this.ideTheme,
+    this.analyticsProvider,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/app.dart:54]: This is going to require a paired change in google3 - we have an internal version of main.dart that initializes DevToolsApp

File: packages/devtools_app/test/integration/dart_cli_profile_test.dart [hunk 4]
```diff
@@ -55,7 +56,8 @@ Future<void> main() async {
       await preferences.init();
       final app = DefaultAssetBundle(
         bundle: _DiskAssetBundle(),
-        child: DevToolsApp(const [], preferences, null),
+        child:
+            DevToolsApp(const [], preferences, null, await analyticsProvider),
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/integration/dart_cli_profile_test.dart:60]: trailing comma

File: packages/devtools_app/lib/src/app.dart (hunk 3 above)
Comment (grouma) [reply packages/devtools_app/lib/src/app.dart:54]: Noted.

**Comment by grouma** [comment]
Merging on red Dart2JS.

**Pull request merged**